controllers/azurecluster_controller.go

```diff
@@ -76,20 +73,23 @@ func NewAzureClusterReconciler(client client.Client, log logr.Logger, recorder r
 
 // SetupWithManager initializes this controller with a manager.
 func (acr *AzureClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options Options) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.AzureClusterReconciler.SetupWithManager")
+	_, log, done := tele.StartSpanWithLogger(ctx,
+		"controllers.AzureClusterReconciler.SetupWithManager",
+		tele.KVP("controller", "AzureCluster"),
+	)
 	defer done()
 
-	log := acr.Log.WithValues("controller", "AzureCluster")
 	var r reconcile.Reconciler = acr
 	if options.Cache != nil {
 		r = coalescing.NewReconciler(acr, options.Cache, log)
 	}
 
+	v6log := log.V(6)
```

Comment:
why are we adding a v(6) here? the predicate functions are already setting different verbosity levels for logging (looks like mostly v(4) and v(6)) 

https://github.com/kubernetes-sigs/cluster-api/blob/3b839fd1c57b9155ef1ecac0380440d73d9ebd7b/util/predicates/generic_predicates.go#L189

Reply:
I think this might be causing the logs to be spammed with all the predicate logs https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1812/pull-cluster-api-provider-azure-e2e/1462803034441519104/artifacts/clusters/bootstrap/controllers/capz-controller-manager/capz-controller-manager-6dd76454d5-tz2cv/manager.log

Reply:
I think you are right. I also think there was an issue with the composite logger not dealing with verbosity level appropriately. I've updated the composite logger to delegate verbosity down to sub loggers.